table/block_based/block_based_table_iterator.cc

```diff
@@ -37,6 +37,66 @@ void BlockBasedTableIterator::SeekImpl(const Slice* target,
                                        bool async_prefetch) {
   // TODO(hx235): set `seek_key_prefix_for_readahead_trimming_`
   // even when `target == nullptr` that is when `SeekToFirst()` is called
+  if (multi_scan_) {
```

Comment:
I suggest moving the multi scan code block in this and in FindBlockForward() to separate functions to reduce complexity